src/index.d.ts

```diff
@@ -6,7 +6,14 @@ export function props(elem: any, data?: PropData): void;
 
 export const emit: (elem: any, eventName: string, eventOptions = {}) => boolean;
 
-export class Component extends HTMLElement {}
+export class Component extends HTMLElement {
+  updatedCallback?(prevProps: { [name: string]: any }): boolean;
+  renderCallback?(): void;
```

Comment:
I had glanced to the tests and there are even arrays being returned for this callback, so I returned the generic `any | undefined` that is used elsewhere as well.

So I am not sure what are all the possibilities.